src/test/integration/org/apache/cassandra/sidecar/common/DelegateTest.java

```diff
@@ -32,31 +36,59 @@
  */
 class DelegateTest
 {
+    private static CassandraAdapterDelegate getCassandraAdapterDelegate(CassandraTestContext context)
+    {
+        CassandraVersionProvider versionProvider = new CassandraVersionProvider.Builder()
+                                                       .add(new Cassandra311Factory(DnsResolver.DEFAULT))
+                                                       .add(new CassandraFactory(DnsResolver.DEFAULT))
+                                                       .build();
+        InstanceMetadata instanceMetadata = context.instancesConfig.instances().get(0);
+        CQLSessionProvider sessionProvider = new CQLSessionProvider(instanceMetadata.host(), instanceMetadata.port(), 1000);
+        CassandraAdapterDelegate delegate = new CassandraAdapterDelegate(versionProvider, sessionProvider, context.jmxClient());
+        return delegate;
+    }
+
     @CassandraIntegrationTest
     void testCorrectVersionIsEnabled(CassandraTestContext context)
     {
-        CassandraVersionProvider provider = new CassandraVersionProvider.Builder().add(new V30()).build();
-        CassandraAdapterDelegate delegate = new CassandraAdapterDelegate(provider, context.session, context.jmxClient);
+        CassandraAdapterDelegate delegate = getCassandraAdapterDelegate(context);
         SimpleCassandraVersion version = delegate.version();
         assertThat(version).isNotNull();
+        assertThat(version.major).isEqualTo(context.version.major);
+        assertThat(version.minor).isEqualTo(context.version.minor);
+        assertThat(version).isGreaterThanOrEqualTo(context.version);
     }
 
     @CassandraIntegrationTest
-    void testHealthCheck(CassandraTestContext context) throws IOException, InterruptedException
+    void testHealthCheck(CassandraTestContext context) throws InterruptedException
     {
-        CassandraVersionProvider provider = new CassandraVersionProvider.Builder().add(new V30()).build();
-        CassandraAdapterDelegate delegate = new CassandraAdapterDelegate(provider, context.session, context.jmxClient);
+        CassandraAdapterDelegate delegate = getCassandraAdapterDelegate(context);
 
         delegate.healthCheck();
 
         assertThat(delegate.isUp()).as("health check succeeds").isTrue();
 
-        context.container.execInContainer("nodetool", "disablebinary");
+        NodeToolResult nodetoolResult = context.cluster.get(1).nodetoolResult("disablebinary");
+        assertThat(nodetoolResult.getRc())
+        .withFailMessage("Failed to disable binary:\nstdout:" + nodetoolResult.getStdout()
+                         + "\nstderr: " + nodetoolResult.getStderr())
+        .isEqualTo(0);
 
-        delegate.healthCheck();
+        for (int i = 0; i < 10; i++)
```

Comment:
Just curious. Why 10?

Reply:
During testing I think it always settled by 1 or 2, but I wasn't sure about how it would work in CI so I wanted to leave some extra time just in case.